lib/httpRequest.ts

```diff
@@ -9,29 +9,22 @@ import { HttpMethod } from "./httpMethod";
 export class HttpRequest {
     private readonly _headers: HttpHeaders;
 
-    constructor(private _httpMethod: HttpMethod, private _url: string, headers: HttpHeaders | RawHttpHeaders, private _body?: string) {
-        if (!this._url) {
-            const urlString: string = (this._url === undefined || this._url === null ? this._url : `"${this._url}"`);
+    /**
+     * Create a new HTTP request using the provided properties.
+     * @param _httpMethod The HTTP method that will be used to send this request.
```

Comment:
This should be httpMethod (remove underscore) as it is not private anymore

Reply:
Done.